packages/app/src/specs/generators/empty/EmptyGenerator.tsx

```diff
@@ -7,5 +7,10 @@ export const EmptyGenerator: SpecGenerator = {
   entry: EmptyGeneratorCardStepOne,
   matches: () => true,
   disabled: () => false,
+  show: (currentProject) => {
+    const specPattern = currentProject?.config.find((item) => item.field === 'specPattern')
+
+    return currentProject?.codeGenGlobs?.component !== '*.vue' || (specPattern && specPattern.from !== 'default')
```

Comment:
We are keying off of the `codeGenGlobs` to determine if it's a vue app (and thus to either show empty or create-from-component). I think it makes more sense to key off of the `framework` that defined in the user's `config.component.devServer`.

Reply:
I agree that this is the way we should do it. I don't see it anywhere in `currentProject` object here though. I wonder if there's something I need to add to the gql query to see that?

Reply:
You might need to add it. I do see an edge case, though. For the "happy path" you can potentially grab the framework from the config, eg `component.devServer.framework`, and then match that up with with the correct one [in this array](https://github.com/cypress-io/cypress/blob/develop/packages/scaffold-config/src/frameworks.ts#L64). It's got a [`codeGenFramework`](https://github.com/cypress-io/cypress/blob/develop/packages/scaffold-config/src/frameworks.ts#L80) key.

The problem will be custom dev servers; we don't really have any way to handle this, since we don't know what framework they are using. I think the eventual fix might be to allow manually configuring how this feature works; someone who wants to configure a dev server themselves likely has the technical skills to configure the template they'd like to use for the scaffold feature, too.

Reply:
https://github.com/cypress-io/cypress/pull/22898#discussion_r939162622